### ISSUE TITLE
Cant install on ubuntu/debian - amazon clean image

### ISSUE THREAD

**Issue by andresmorago** [ab77/netflix-proxy#272]
hello
im getting stuck at the end of the installation. could you please provide me some ideas?

```
ubuntu@ip-172-26-14-8:/opt/netflix-proxy$ sudo curl -L https://github.com/ab77/netflix-proxy/archive/latest.tar.gz | sudo tar xz --strip-components=1
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   126    0   126    0     0    618      0 --:--:-- --:--:-- --:--:--   620
100 3119k    0 3119k    0     0  4024k      0 --:--:-- --:--:-- --:--:-- 4024k


ubuntu@ip-172-26-14-8:/opt/netflix-proxy$ sudo  ./build.sh -c 186.81.xxx.xx
Command line is not complete. Try option "help"
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    14  100    14    0     0     62      0 --:--:-- --:--:-- --:--:--    62
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

curl: (6) Could not resolve host: ifconfig.co
resolved params: clientip= client_ipv4=1 client_ipv6=1 ipaddr=172.26.14.8 ipaddr6= extip=34.xxx.xxx.xx extip6=
cmd: ./build.sh -r=1 -b=0 -s= -z=0 -n=1 -u= -p [secret] -k [secret]

* enabling DNS recursion...								[ OK ]
* unable to resolve and authorise client ip...
* adding IPv4 iptables rules... 								[ OK ]
* adding IPv6 iptables rules..								[ OK ]
* creating Docker and sniproxy configuration templates...		[ OK ]
* checking IPv6 connectivity...										[fail]
* configuring sniproxy and Docker...						[ OK ]
* installing iptables|netfilter-persistent service...				[ OK ]
* socialising Docker with iptables-persistent service...			[ OK ]
* saving iptables rules...									[ OK ]
* creating zones.override from template...					[ OK ]
* updating db.override with extip=34.225.102.65 extip6= and date=20170731...		[ OK ]
* installing python-pip and docker-compose...				[ OK ]
* configuring netflix-proxy-admin backend...					[ OK ]
* configuring netflix-proxy-admin reverse-proxy...						[ OK ]
* creating cron scripts...											[ OK ]
* creating and starting Docker containers...							[ OK ]
/sbin/init: unrecognized option '--version'
* configuring systemd...											[fail]
* reloading ipables rules...										[ OK ]
* no specific steps to execute for Ubuntu at this time...					[ OK ]
* testing DNS...

```

**Comment by ab77** [comment]
I've made some changes to the master branch, but forgot to update the release. Try now...

**Comment by andresmorago** [comment]
hello ab
thanks for the feedback. i will try right away.

could you please confirm if this code is ok? will it work for both ubuntu and debian?

```
sudo apt-get update && sudo apt-get -y install vim dnsutils curl sudo git 
sudo curl -sSL https://get.docker.com/ | sudo sh 
sudo mkdir -p /opt/netflix-proxy && cd /opt/netflix-proxy 
sudo curl -L https://github.com/ab77/netflix-proxy/archive/latest.tar.gz | sudo tar xz --strip-components=1 
sudo  ./build.sh
```

**Comment by andresmorago** [comment]
hello ab
no luck so far. same issue occurs. :(

**Comment by ab77** [comment]
Try commenting out all of the tests at the bottom. Need to work out where it hangs. I tested on a DigitalOcean Docker 16.04 image with IPv6 and it worked, so it's going to be something specific to your environment.

**Comment by andresmorago** [comment]
hi ab
i tried again today on a fresh debian instance. i get the following errors and progress stops on the middle.
please note the erroes that appear

```
admin@ip-172-26-2-26:/opt/netflix-proxy$ sudo  ./build.sh
Command line is not complete. Try option "help"
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    14  100    14    0     0     56      0 --:--:-- --:--:-- --:--:--    56
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (6) Could not resolve host: ifconfig.co
resolved params: clientip= client_ipv4=1 client_ipv6=1 ipaddr=172.26.2.26 ipaddr6= extip=34.xxx.xxx.xx extip6=
cmd: ./build.sh -r=1 -b=0 -s= -z=0 -n=1 -u= -p [secret] -k [secret]

[ ok ] enabling DNS recursion...done.
[ ok ] unable to resolve and authorise client ip...
[....] adding IPv4 iptables rules...done.
[ ok ] adding IPv6 iptables rules...done.
[ ok ] creating Docker and sniproxy configuration templates...done.
[FAIL] checking IPv6 connectivity...failed.
[ ok ] configuring sniproxy and Docker...done.
[ ok ] installing iptables|netfilter-persistent service...done.
[ ok ] socialising Docker with iptables-persistent service...done.
[ ok ] saving iptables rules...done.
[ ok ] creating zones.override from template...done.
[ ok ] updating db.override with extip=34.xxx.xxx.xx extip6= and date=20170813...done.
[ ok ] installing python-pip and docker-compose...done.
[ ok ] configuring netflix-proxy-admin backend...done.
[ ok ] configuring netflix-proxy-admin reverse-proxy...done.
[ ok ] creating cron scripts...done.
[....] creating and starting Docker containers...
admin@ip-172-26-2-26:/opt/netflix-proxy$
```

**Comment by ladigb** [comment]
Hi
its look like problem with port 53 try remove dnsmasq should help

**Comment by ab77** [comment]
What do you have in the build log?

**Comment by ardadem** [comment]
same problem guys, i tried remove dnsmasq.

cmd: ./build.sh -r=1 -b=0 -s= -z=0 -n=1 -u= -p [secret] -k [secret]

 * enabling DNS recursion...                                                                                     [ OK ] 
 * authorising clientip=88.250.173.177 on iface=eth0...                                                          [ OK ] 
 * adding IPv4 iptables rules...                                                                                 [ OK ] 
 * adding IPv6 iptables rules...                                                                                 [ OK ] 
 * creating Docker and sniproxy configuration templates...                                                       [ OK ] 
 * checking IPv6 connectivity...                                                                                         * enabling sniproxy IPv6 priority...                                                                            [ OK ] 
 * installing sipcalc...                                                                                         [ OK ] 
 * automatically calculating IPv6 subnet...                                                                      [ OK ] 
 * enabling Docker IPv6 dual-stack support with fixed-cidr-v6=2a03:b0c0:0001:00d0:0000:0000:27ee:3008/125...     [ OK ] 
 * adding IPv6 iptables rules...                                                                                 [ OK ] 
 * installing iptables|netfilter-persistent service...                                                           [ OK ] 
 * saving iptables rules...                                                                                      [ OK ] 
 * creating zones.override from template...                                                                      [ OK ] 
 * updating db.override with extip=178.62.57.96 extip6= and date=20170922...                                     [ OK ] 
 * installing python-pip and docker-compose...                                                                   [ OK ] 
 * configuring netflix-proxy-admin backend...                                                                    [ OK ] 
 * configuring netflix-proxy-admin reverse-proxy...                                                              [ OK ] 
 * creating cron scripts...                                                                                      [ OK ] 
 * creating and starting Docker containers...                                                                           root@mycloud:/opt/netflix-proxy#

**Comment by Carylgithub** [comment]
Hello,
Same issue....at the same point...install on scaleway....

Any idea ?

**Comment by ab77** [comment]
Yes, have a look in netflix-proxy.log.

**Comment by Carylgithub** [comment]
Hello,

Thank you for your quick response...
I put the results of the installation below and the netflix-proxy.log file.

As a reminder, I tried the installation on a scaleway droplet in France with IPV6 activated and without tunnelbloker.....

Thank you in advance!

```
Docker 1.12.2 is running using the '' storage driver.
Installed tools: docker-compose, nsenter, gosu and pipework are installed.
Last login: Sun Dec  3 12:50:53 2017 from 51.15.***.***
root@stream:~# git clone https://github.com/ab77/netflix-proxy /opt/netflix-proxy\
>   && cd /opt/netflix-proxy\
>   && ./build.sh
fatal: destination path '/opt/netflix-proxy' already exists and is not an empty directory.
root@stream:~# rm /opt/netflix-proxy\
> 
rm: cannot remove '/opt/netflix-proxy': Is a directory
root@stream:~# && cd /opt/netflix-proxy\
-bash: syntax error near unexpected token `&&'
root@stream:~#   && ./build.sh
-bash: syntax error near unexpected token `&&'
root@stream:~# cd /opt/netflix-proxy\
>   && ./build.sh
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    13  100    13    0     0    221      0 --:--:-- --:--:-- --:--:--   224
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    27  100    27    0     0    531      0 --:--:-- --:--:-- --:--:--   540
testing IPv4: '105.154.***.***'
testing IPv6: '105.154.***.***'
resolved params: clientip=105.154.***.*** client_ipv4=0 client_ipv6=1 ipaddr=10.2.209.*** ipaddr6= extip=51.15.***.** extip6=2001:bc8:4400:***::**:***
cmd: ./build.sh -r=1 -b=0 -s= -z=0 -n=1 -u= -p [secret] -k [secret]

 * log diagnostics info...                                               [ OK ] 
 * enabling DNS recursion...                                             [ OK ] 
 * authorising clientip=105.154.193.91 on iface=eth0...                  [ OK ] 
 * adding IPv4 iptables rules...                                         [ OK ] 
 * adding IPv6 iptables rules...                                         [ OK ] 
 * creating Docker and sniproxy configuration templates...               [ OK ] 
 * checking IPv6 connectivity...                                                 * enabling sniproxy IPv6 priority...                                    [ OK ] 
 * installing sipcalc...                                                 [ OK ] 
 * automatically calculating IPv6 subnet...                              [ OK ] 
 * enabling Docker IPv6 dual-stack support with fixed-cidr-v6=1:0/125... [ OK ] 
 * adding IPv6 iptables rules...                                         [ OK ] 
 * installing iptables|netfilter-persistent service...                   [ OK ] 
 * saving iptables rules...                                              [ OK ] 
 * creating zones.override from template...                              [ OK ] 
 * updating db.override with extip=51.15.214.34 extip6=2001:bc8:4400:2100::2e:a09 and date=20171203...                                                   [ OK ] 
 * installing python-pip and docker-compose...                           [ OK ] 
 * configuring netflix-proxy-admin backend...                            [ OK ] 
 * configuring netflix-proxy-admin reverse-proxy...                      [ OK ] 
 * creating cron scripts...                                              [ OK ] 
 * creating and starting Docker containers...                                   root@stream:/opt/netflix-proxy# 
```

**netflix-proxy.log**

```
Requirement already satisfied: ipaddress>=1.0.16; python_version < "3.3" in /usr/local/lib/python2.7/dist-packages (from docker-py<2.0,>=1.10.3->docker-compose)
Requirement already satisfied: docker-pycreds>=0.2.1 in /usr/local/lib/python2.7/dist-packages (from docker-py<2.0,>=1.10.3->docker-compose)
Requirement already satisfied: functools32; python_version == "2.7" in /usr/local/lib/python2.7/dist-packages (from jsonschema<3,>=2.5.1->docker-compose)
Requirement already satisfied: dnspython in /usr/local/lib/python2.7/dist-packages (from -r /opt/netflix-proxy/auth/requirements.txt (line 1))
Requirement already satisfied: web.py in /usr/local/lib/python2.7/dist-packages (from -r /opt/netflix-proxy/auth/requirements.txt (line 2))
Requirement already satisfied: passlib in /usr/local/lib/python2.7/dist-packages (from -r /opt/netflix-proxy/auth/requirements.txt (line 3))
proxy / localhost:43867 {
    except /static
    header_upstream Host {host}
    header_upstream X-Forwarded-For {remote}
    header_upstream X-Real-IP {remote}
    header_upstream X-Forwarded-Proto {scheme}
}
Couldn't connect to Docker daemon - you might need to run `docker-machine start default`.
```

**Comment by ab77** [comment]
Redeploy your OS and try again with the latest version. If that doesn't work, see if Docker is installed first.

**Comment by Carylgithub** [comment]
Thank you for that information.
It was a fresh install on a latest scaleway docker image, so docker was installed first....
it's weird.....It seems to be a common mistake...no ?

(I did the installation on a droplet digital ocean without any problem. It's the scaleway supplier that seems to be blocking the installation.)

**Comment by ab77** [comment]
Have you tried again now, using the latest code?

**Comment by Carylgithub** [comment]
Ah? Did you update the code today? I did the installation yesterday:
Sun Dec 3 12:50:53 2017 with this git clone https://github.com/ab77/netflix-proxy

**Comment by ab77** [comment]
Check commit dates.

**Comment by Carylgithub** [comment]
The installation has taken another step forward but it still doesn't go through... here is the installation information

```
* creating and starting Docker containers...                            [ OK ] 
 * configuring init system...  
/sbin/init: unrecognized option '--version'
                                                                         [ OK ]
 * reloading ipables rules...                                            [ OK ] 
 * testing DNS...
 root@stream:/opt/netflix-proxy# 
```

and the end of the log file

```
Created symlink from /etc/systemd/system/multi-user.target.wants/netflix-proxy-admin.service to /lib/systemd/system/netflix-proxy-admin.service.
Created symlink from /etc/systemd/system/multi-user.target.wants/systemd-networkd.service to /lib/systemd/system/systemd-networkd.service.
Created symlink from /etc/systemd/system/sockets.target.wants/systemd-networkd.socket to /lib/systemd/system/systemd-networkd.socket.
Created symlink from /etc/systemd/system/network-online.target.wants/systemd-networkd-wait-online.service to /lib/systemd/system/systemd-networkd-wait-online.service.
 * Loading netfilter rules...
run-parts: executing /usr/share/netfilter-persistent/plugins.d/15-ip4tables start
run-parts: executing /usr/share/netfilter-persistent/plugins.d/25-ip6tables start
   ...done.
/opt/netflix-proxy/scripts/functions: line 79: -4: command not found
Failure! Retrying in 5..
/opt/netflix-proxy/scripts/functions: line 79: -4: command not found
Failure! Retrying in 10..
/opt/netflix-proxy/scripts/functions: line 79: -4: command not found
Failure! Retrying in 20..
You've failed me for the last time! (-4 +time=5 netflix.com @51.15.***.***)
/opt/netflix-proxy/scripts/functions: line 79: -4: command not found
```

**Comment by ab77** [comment]
Are you following the instructions? cURL is missing.

```apt-get update\
  && apt-get -y install vim dnsutils curl sudo git\
  && curl -sSL https://get.docker.com/ | sh\
  && mkdir -p ~/netflix-proxy\
  && cd ~/netflix-proxy\
  && curl -L https://github.com/ab77/netflix-proxy/archive/latest.tar.gz | tar xz --strip-components=1\
  && ./build.sh
```

**Comment by Carylgithub** [comment]
maybe not....I'm retrying from an other fresh install....thanks

**Comment by Carylgithub** [comment]
2 days that I'm on it...always failed....
Sorry I'm giving up... I'm tired, it's too complicated....I go back to my VPN ;-)
Thanks anyway....

**Comment by ab77** [comment]
FWITW, I've tested on Scaleway and it worked fine.

**Comment by Carylgithub** [comment]
It's too complicated for me.
When I use an image with a preinstalled docker
("Docker 1.12.2 is running using the' aufs' storage driver.
Installed tools: docker-composer, nsenter, gosu and pipework are installed.")
I can't understand what tutorial I have to use.
I tried by placing in ~/netflix-proxy
But it creates mistakes like that:
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
root@stream: ~# packet_write_wait: Connection to 51.15.***.*** port 22: Broken pipe
... etc, etc... etc....
In short, it is impossible to find the right package....

You've been very specific in your tutorials but I couldn't install it... It's my fault, I'm sorry.....

**Comment by ab77** [comment]
Try the Docker image (Scaleway -> Create Server -> ImageHub -> Docker).

Worked fine for me.

**Comment by Carylgithub** [comment]
Yes I did it 10 times ;-) (at least ;-)
But could you give me the right commands after I was connected on my new image hub ?

these don't work for me
mkdir -p ~/netflix-proxy\
  && cd ~/netflix-proxy\
  && curl -L https://github.com/ab77/netflix-proxy/archive/latest.tar.gz | tar xz --strip-components=1\
  && ./build.sh

**Comment by ab77** [comment]
```mkdir -p ~/netflix-proxy\
  && cd ~/netflix-proxy\
  && curl -L https://github.com/ab77/netflix-proxy/archive/latest.tar.gz | tar xz --strip-components=1\
  && ./build.sh```

**Comment by Carylgithub** [comment]
I retry.....just in case....thank you

**Comment by Carylgithub** [comment]
After that:

```
mkdir -p ~/netflix-proxy
&& cd ~/netflix-proxy\
  && curl -L https://github.com/ab77/netflix-proxy/archive/latest.tar.gz | tar xz --strip-components=1\
  && ./build.sh

```
It stop at the end of this lines 

```
 * checking if cURL is installed...                                      [ OK ] 
 * checking if Docker is installed...                                    [ OK ] 
 * checking if sudo is installed...                                      [ OK ] 
 * checking if dig is installed...                                            
 root@stream:~/netflix-proxy# 
```

**Comment by ladigb** [comment]
Hi
 you can try some other way to get netflix working vpn working, install some openvpn server in amazon  can be pritunl its easy, and the you will need some raspberry and install as vpn router, https://gist.github.com/superjamie/ac55b6d2c080582a3e64  

and its work perfect with netflix 

P.s Amazon image have build it own firewall i think you need to first remove it check if  dnsmasq is not installed coz then is block port 53 and cant install docker

hop it help

**Comment by ab77** [comment]
@andresmorago you are obviously missing the `dig` command. Google on how to install it.

**Comment by Carylgithub** [comment]
@ladigb Yes, it's an excellent idea but for the moment openvpn is reducing the bandwidth too much (from 100 to 15 MB).
A very powerful router is needed to achieve a minimum throughput of 20/45 MB....
Is the raspberry strong enough? 

That's why I'm moving towards smartdns... but it's not very simple.....

**Comment by ab77** [comment]
We've done tests on RPi3 with no encryption/auth and 10+MBit/s is definitely possible with the image at http://unzoner.com and a reasonably fast SDCard. In fact, I've seen almost 20Mbit/s.

But in general yes, don't expect  GigE speeds from a device that fits into the palm of your hand, at least not for a while.

If you have a suitably sized router board with DD-WRT, this is for you: http://unzoner.com/#dd-wrt.

-- ab1

**Comment by Carylgithub** [comment]
A better router is also a good solution 👍 
I've just read this.....its depends on CPU

Tested by the writer :
600 Mhz = 8 – 10 Mbit/s
800 Mhz = 15 – 25 Mbit/s
1000 Mhz = 20 – 35 Mbit/s
1400 Mhz = 25 – 45 Mbit/s

http://le-routeur-wifi.com/les-meilleurs-routeurs-vpn-en-2017/

**Comment by ladigb** [comment]
Hi
i have just check, and i  have broadband  download 37mb/s  upload 8mb  and speed.ent show me download 34.90mb/s upload 7.89mb/s and i think i use encryption 128 and rapi3

**Comment by Carylgithub** [comment]
@ladigb very good performances ! I'll try this....

**Comment by ladigb** [comment]
also in link what i give for setup openvpn rouer you can find comment tells:

"I did try my setup with a RPi2, PRi3 and ODROID C2 before going to my current virtualbox debian setup. My downlink speed maxes out at 60Mb/s. I was able to max the link with the ODROID C2 running as the gateway. However the VPN tunnel kept interrupting and not recovering for some unknown reason. I don't remember the exact thruput I got with the RPi3, but it was somewhere between 30-50Mb/s"

**Comment by Carylgithub** [comment]
Thank you @ab77 , thank you very much! after correction of the DIG command Scaleway installation works perfectly!

Time for tests! Apple TV 4K and Molotov. tv (In france)......

Congratulations again and thank you for your support

**Comment by ab77** [comment]
Very good.
